internal/provider/resource_group.go

```diff
@@ -116,7 +117,12 @@ func (r groupResource) Read(ctx context.Context, req tfsdk.ReadResourceRequest,
 
 	group, err := r.provider.client.ReadGroup(data.Id.Value)
 	if err != nil {
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read group, got error: %s", err))
+		if strings.Contains(err.Error(), statusNotFound) {
```

Comment:
String comparison for error handling is generally considered to be a fragile approach, because the error message format can easily change in the future. A more robust way to handle HTTP errors would be to use the HTTP status code returned by the server.

Can we please consider doing something like:
1. Return the HTTP Response in doRequest: Modify the doRequest method to return both the HTTP response and any error.
Instead of `return body, err` let's return `return body, res, err`
3. In the RequestAndMap function, you can check the HTTP status code and return a custom error type if it's a 404.
Something like
`if resp.StatusCode == http.StatusNotFound {
		return NotFoundError{}
	}`
4. Handle the Custom Error in Read: In the Read method, you can use a type assertion to check if the error is of the custom error type.
Something like `if _, ok := err.(NotFoundError); ok {`
Instead of `if strings.Contains(err.Error(), statusNotFound) {`

Reply:
I refactored the code. Can you please take a look?

Reply:
Looks great now 👍